### ISSUE TITLE
Randomly picks wrong locale

### ISSUE THREAD

**Issue by mavenastic** [globalize/globalize#462]
Hi,

I am using `globalize` v5.0.1 with `globalize-accessors` v0.2.1 and `rails` v4.2.1. Here is my configuration:

```
config.i18n.available_locales = [:fr, :en]
config.i18n.default_locale = :fr
config.i18n.fallbacks = true
Globalize.fallbacks = { fr: [:fr, :en], en: [:en, :fr] }
```

When I18n.locale is set to `:en` (from a subdomain in the URL), I get the French value of `@model.attribute` in my templates. This issue appears randomly and only affects model attributes (translations for the UI are displayed in English) which indicates I18n.locale is set correctly. If I use `@model.send("attribute_#{I18n.locale}".to_sym)`, I do get the correct value. Restarting `unicorn` resolves this issue. I am using a t2.micro AWS EC2 instance (the smallest one) and nor the memory or the disk seem full when this happens. I have another staging server which is identical to the first one and where I cannot reproduce the issue.

How can I troubleshoot this problem? Is this a bug?

Thanks in advance!

**Comment by simi** [comment]
Hello, can you try master branch where where [this](https://github.com/globalize/globalize/pull/420) is merged? Probably it is good time to release 5.1.0 also.

**Comment by mavenastic** [comment]
Thank you @simi. I just updated my Gemfile. Will keep you posted.

**Comment by simi** [comment]
@mavenastic also inspect [CHANGELOG.md](https://github.com/globalize/globalize/blob/5db91fc97e5c5ab4ee51742eb6c462bc394c1020/CHANGELOG.md#510-unreleased) for another changes. But I think #420 can be backported to 5.0 also, if you will have problems with another changes.

**Comment by mavenastic** [comment]
Thanks @simi, will try!

**Comment by diegosteiner** [comment]
+1 for the release of 5.1.0

Restarting the webserver (puma) fixed the issue temporarly for me too.

**Comment by parndt** [comment]
@simi I am happy to have 5.1.0 if you are?

**Comment by cimm** [comment]
There is a `beta` [release](https://github.com/globalize/globalize/blob/master/CHANGELOG.md#510-unreleased) that should have this issue fixed if I am not mistaken so please test [5.1.0.beta1](https://rubygems.org/gems/globalize/versions/5.1.0.beta1).
